services/consensusalgo/leanhelixconsensus/block_provider.go

```diff
@@ -0,0 +1,122 @@
+package leanhelixconsensus
+
+import (
+	"context"
+	"github.com/orbs-network/lean-helix-go"
+	lhprimitives "github.com/orbs-network/lean-helix-go/primitives"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/crypto/logic"
+	"github.com/orbs-network/orbs-network-go/crypto/signature"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/consensus"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/orbs-network/orbs-spec/types/go/services/handlers"
+)
+
+func (s *service) HandleBlockConsensus(ctx context.Context, input *handlers.HandleBlockConsensusInput) (*handlers.HandleBlockConsensusOutput, error) {
+	panic("implement validate block consensus (call the lib)")
+}
+
+func (s *service) HandleLeanHelixMessage(ctx context.Context, input *gossiptopics.LeanHelixInput) (*gossiptopics.EmptyOutput, error) {
+	panic("implement me")
+}
+
+func (s *service) getLastCommittedBlock() (primitives.BlockHeight, *protocol.BlockPairContainer) {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+
+	if s.lastCommittedBlockUnderMutex == nil {
+		return 0, nil
+	}
+	return s.lastCommittedBlockUnderMutex.TransactionsBlock.Header.BlockHeight(), s.lastCommittedBlockUnderMutex
+}
+
+func (s *service) RequestNewBlock(ctx context.Context, blockHeight lhprimitives.BlockHeight) (leanhelix.Block, error) {
+
+	_lastCommittedBlockHeight, _lastCommittedBlock := s.getLastCommittedBlock()
+	s.logger.Info("generating new proposed block", log.BlockHeight(_lastCommittedBlockHeight+1))
+
+	// get tx
+	txOutput, err := s.consensusContext.RequestNewTransactionsBlock(ctx, &services.RequestNewTransactionsBlockInput{
```

Comment:
context probably needs to have a deadline

Reply:
Added, though not sure who should handle a timeout, and how.
We can tweak it later, or remove it altogether.